app/components/layout/sidebar/sidebar.tsx

```diff
@@ -42,8 +86,51 @@ export default function Sidebar() {
             className="h-full"
           />
         </Link>
-        <MenuButton />
+        <div className="flex items-center space-x-4">
+          <button
+            onClick={() => setShowScanner(true)}
+            title="Scan QR Code"
+            className="relative flex items-center justify-center px-2 transition"
+          >
+            <BarCodeIcon className="h-6 w-6 text-gray-500" />
+          </button>
+          <MenuButton />
+        </div>
       </header>
+      {showScanner && (
+        <div className="relative">
+          <QRCodeScanner
+            onScan={(data) => handleQRScan(data)}
+            onClose={() => setShowScanner(false)}
+          />
+          {/* Add a close button on top right corner of the scanner */}
+          <button
+            className="absolute right-2 top-2 flex h-8 w-8 items-center justify-center rounded-full bg-red-500 text-white"
+            onClick={handleScannerClose}
+            title="Close Scanner"
+          >
+            &times; {/* This is an "X" character */}
+          </button>
+        </div>
+      )}
+
+      {scannedSuccessfully && (
+        <div className="my-4 flex items-center justify-center rounded border border-green-300 bg-green-100 p-4 text-green-700">
+          Assets scanned successfully...
+        </div>
+      )}
+
+      {errorMessage && (
```

Comment:
This is fine but I would like to hear a bit more about it. What type of errors are we handling here? Can you show some examples of how the error will look?

Reply:
this will show error if qr code data validation fails

Reply:
Uploading Screen Recording 2023-09-18 at 6.01.22 PM.mov…